beacon-chain/sync/pending_blobs_queue.go

```diff
@@ -0,0 +1,147 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed"
+	statefeed "github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed/state"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
+	eth "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+)
+
+// processPendingBlobs listens for state changes and handles pending blobs.
+func (s *Service) processPendingBlobs() {
+	s.waitForChainStart()
+
+	eventFeed := make(chan *feed.Event, 1)
+	sub := s.cfg.stateNotifier.StateFeed().Subscribe(eventFeed)
+	defer sub.Unsubscribe()
+
+	// Initialize the cleanup ticker at 5s and 10s mark. The node is less busy that time.
+	cleanupTicker := slots.NewSlotTickerWithIntervals(s.cfg.chain.GenesisTime(), []time.Duration{5, 10})
+
+	for {
+		select {
+		case <-s.ctx.Done():
+			return
+		case <-sub.Err():
+			return
+		case e := <-eventFeed:
+			s.handleEvent(s.ctx, e)
```

Comment:
We're doing some time-consuming work downstream from `handleEvent`, including making RPC calls to peers. I see you are buffering the channel receiving the events from the state feed subscription (`eventFeed := make(chan *feed.Event, 1)`), but the /second/ time something tries to write to state feed, if we're still doing the time-consuming work, we will block the process trying to write to state feed. Is that a concern here?

Reply:
`handleNewBlockEvent` is fast, I don't see many concerns here. In most cases, `pendingBlobSidecars` is empty anyway